openml/tasks/functions.py

```diff
@@ -224,36 +225,90 @@ def get_task(task_id):
         raise ValueError("Task ID is neither an Integer nor can be "
                          "cast to an Integer.")
 
-    xml_file = os.path.join(_create_task_cache_dir(task_id),
-                            "task.xml")
+    tid_cache_dir = _create_task_cache_dir(task_id)
 
     with lockutils.external_lock(
             name='datasets.functions.get_dataset:%d' % task_id,
             lock_path=os.path.join(config.get_cache_directory(), 'locks'),
     ):
         try:
-            with io.open(xml_file, encoding='utf8') as fh:
-                task = _create_task_from_xml(fh.read())
+            task = _get_task_description(task_id)
```

Comment:
I don't get why the logic changed so much here...

Reply:
To make it look like getting a dataset. Things are now in their own little functions.

Reply:
hm ok. I can try to go through the details. I mean it looks ok but I haven't looked through the details.

Reply:
The tests are passing and I just shuffled some lines, so don't think there's something weird going on here.